test_runner/regress/test_compute_metrics.py

```diff
@@ -0,0 +1,19 @@
+from fixtures.neon_fixtures import NeonEnv
+
+
+def test_compute_metrics(neon_simple_env: NeonEnv):
+    """
+    Test compute metrics, exposed in the neon_backend_perf_counters and
+    neon_perf_counters views
+    """
+    env = neon_simple_env
+    endpoint = env.endpoints.create_start("main")
+
+    conn = endpoint.connect()
+    cur = conn.cursor()
+
+    # We don't check that the values make sense, this is just a very
+    # basic check that the server doesn't crash or something like that.
+    cur.execute("CREATE EXTENSION neon")
+    cur.execute("SELECT * FROM neon_perf_counters")
+    cur.execute("SELECT * FROM neon_backend_perf_counters")
```

Comment:
Surely these would be all zero?

Should the getpage wait thresholds be checked against pageserver getpage metric buckets that they are in sync?

Reply:
> Surely these would be all zero?

You mean, in this test, because it hasn't done anything? No, even just connecting to Postgres requires fetching some pages for the system catalogs that are needed for authentication.

> Should the getpage wait thresholds be checked against pageserver getpage metric buckets that they are in sync?

That would be difficult to automate. There's always some network latency, so the latencies measured from the compute are expected to be somewhat higher. But they can also be lower, if prefetching is effective; these reported wait times start the clock when we enter the smgrread() call and know that we need to read a page. If a prefetch request has been issued for the page earlier, the response might already be in flight.

Comparing the values as measured from pageserver is a very valuable thing to do manually, though. It can tell a lot about network latency and how effective the prefetching is.

Reply:
> For the second question, I meant to ask about asserting that the bucket thresholds are the same, not any of their contents.

Gotcha. Not sure it matters if they're the same or not. And I'm not sure what the thresholds should be, how much granularity do we need. It's pretty cheap to have more buckets in the compute side, we can also decide to export just some of them in the metrics exporter.